File: tests/view_mamullin.php
<?php include("func.inc");

// AUTHOR: Michael Mullin
// DATE LAST MODIFIED: DEC 1/ 2004
// Script Name: view_mamullin.php
//
// This script shows the items contained in the database
// Logically deleted items are not shown

session_start();

import_request_variables("g","g_");
import_request_variables("p","p_");
import_request_variables("c","c_");

if ( $c_PHPSESSID == logged($_SESSION['user']) && isset($c_PHPSESSID) ) { // check to see if we are logged on
	head("View");
	topmenu($_SESSION['user'],"View");
	$dblog = my_sql_login();
	$result = @mysql_query("SELECT * FROM productX") or die("Could not run query from Database");
	$count = 0;

	echo "<form method=\"POST\" action=\"delete_mamullin.php\">";
	while ($row = mysql_fetch_array($result)) { // while we have rows to show, show them
		if( $row[deleted] != 1 ) { // make sure we do not show deleted items
			form($row[pid],$row[name],$row[noh],$row[rol],
				$row[catagory],$row[model],$row[price],
				$row[description],$count);
			++$count;
		}
	}
	mysql_free_result($result);
	mysql_close($dblog);
	bottom();
} else { // we are not successfully logged on
	header("Location: login_mamullin.php");
}
?>

<?php function form($pid,$name,$noh,$rol,$catagory,$model,$price,$description,$count)
{ // the form is a collection of tables containing the items data
	$background = "#00BB00;";
	$colour = "";
	$link = "";
	if( $rol > $noh ) { // number on hand is less than re-order level
		$link = "dark";
		$background = "#226622;";
		$colour = "color:#FFFFFF;";
	}

	switch($catagory)
	{ // our catagory drop down box will contain full words
	case "OTH":
		$catagory = "Other";
		break;
	case "Shp":
		$catagory = "Ship";
		break;
	case "prb":
		$catagory = "Probe";
		break;
	case "bwp":
		$catagory = "Balistic Weapon";
		break;
	case "lwp":
		$catagory = "Laser/Phaser Weapon";
		break;
	case "cht":
		$catagory = "Star Chart";
		break;
	case "stn":
		$catagory = "Space Station";
		break;
	case "sat":
		$catagory = "Satelite";
		break;
	default:
		$catagory = "UNKNOWN";
	}
?>
<table border="1" style="<?=$colour ?>text-align:left;background-color:<?=$background ?>
	width:100%;">
<tr>
	<td style="width:15%"><strong>Product ID:</strong>
	<a href="addmod_mamullin.php?pid=<?=$pid ?>" class="<?=$link ?>"> <?=$pid ?></a></td>
	<td style="width:25%"><strong>Product Name:</strong><?=$name ?></td>
	<td style="width:18%"><strong>Current Supply:</strong><?=$noh ?></td>
	<td style="width:18%"><strong>Re-order Level:</strong><?=$rol ?></td>
	<td style="width:24%"><strong>Catagory:</strong><?=$catagory ?></td>
</tr>
<tr>
	<td><strong>Model:</strong><?=$model ?></td>
	<td><strong>Price:</strong>$<?php printf("%.2f",$price) ?></td>
<?php
if( $noh >= $rol )
{ // put the positive message
?>
	<td colspan="2"><strong> We have adaquate supply</strong></td>
<?php
}else { // put the negative message
?>
	<td colspan="2"><strong> We need more of this item</strong></td>
<?php
}
?>
	<td><strong>Mark for Delete</strong>
	<input name="<?=$count?>" type="checkbox" value="<?=$pid?>" /></td>
</tr>
<tr>
	<td colspan="2" style="width:20%"><strong>Description:</strong>
	<?=$description ?>
	</td>
</tr>
</table><br />
<?php
}
?>

<?php function bottom()
{ // the bottom of the page
?>
<input type="submit" name="submit" value="Delete" />
<input type="reset" name="reset" value="Reset" />
</form>
<a href="logout_mamullin.php">Log Out</a>
<a href="source.html">Source Code</a>
<?php
}
?>

File: tests/view_mamullin.pl
#!/usr/bin/perl

# MAIN.PL
# AUTHOR: MICHAEL MULLIN
# DATE LAST MODIFIED: November 12/ 2004

use warnings;
use strict;
use DBI;
use CGI ":standard";
use CGI::Pretty;
use CGI::Carp "fatalsToBrowser";

require "subs_mamullin.lib";

if( &logged_on() ) {

	# get auth cookie to pass value into top menu (username display)
	my $cookie = cookie("auth");
	my @auth = split(":", $cookie);

	# connect to database
	my $datah = DBI->connect("DBI:mysql:database=b3201a21;host=zenit-db",
		"b3201a21", "quest1", {RaiseError=>1});

	# get all the non-deleted records for display
	my $sql = "SELECT * FROM product WHERE deleted = 0 ORDER BY pid";
	my $stmth = $datah->prepare($sql);
	$stmth->execute();

	# put the necessary header and top menu
	&heading("Michael\'s Space Emporium : View/Update");
	print "<body>\n";
	&top($auth[0],"View");


	print "<table border=\"1\">";

	my $line;
	my $count = 0;
	my $addmod;
	my $linkColor = "dark";
	my $price;
	my $catagory;
	my $supply;

	print "<form action=\"delete_mamullin.pl\" method=\"post\">";

	# if we have data to display, do so
	while( $line = $stmth->fetchrow_hashref ) {

		# get the full catagory word from three letter database store		
		$catagory = &getCat($line->{catagory});

		# make sure the price is formatted correctly
		$price = sprintf("%.02lf", $line->{price});
		$price = "\$" . $price;

		# make sure that if we chose to modify a record, we select the correct
		# record via get principals
		$addmod = "addmod_mamullin.pl?pid=" . $line->{pid};

		# mark the record if number on hand is less than re-order level
		if( int($line->{rol}) <= int($line->{noh}) ) {
			print "<table border=\"1\" "
				. "style=\"text-align:left;background-color:#00BB00;"
				. "width:100%;\">";
			$supply = "We have adaquate supply";
			$linkColor = "";
		}else {
			print "<table border=\"1\" style=\"color:#FFFFFF;"
				. "text-align:left;background-color:#226622;width:100%;\">";
			$supply = "We need more of this item";
			$linkColor = "dark";
		}
		# print out the table containing info
###START HERE###
		print<<END;
		<tr>
			<td style="width:15%"><strong>Product ID:</strong>
			<a href="$addmod" class="$linkColor"> $line->{pid} </a></td>
				<td style="width:25%"><strong>Product Name:</strong> $line->{name}</td>
			<td style="width:18%"><strong>Current Supply:</strong> $line->{noh}</td>
			<td style="width:18%"><strong>Re-order Level:</strong> $line->{rol}</td>
			<td style="width:24%"><strong>Catagory:</strong> $catagory </td>
		</tr>
		<tr>
			<td><strong>Model:</strong> $line->{model}</td>
			<td><strong>Price:</strong> $price</td>
			<td colspan=\"2\"><strong> $supply</strong></td>
			<td><strong>Mark for Delete</strong>
				<input name="$count" type="checkbox" value="$line->{pid}" />
			</td>
		</tr>
		<tr>
			<td colspan=\"2\" style=\"width:20%\"><strong>Description:</strong>
				$line->{description}</td>
		</tr>
		</table><br />
END
###END HERE###
		$count++;

	} #end of while
	# clicking delete will logically delete all marked records
	# reset will un-mark any marked records
	print "<input type=\"submit\" name=\"submit\" value=\"Delete\" />\n";
	print "<input type=\"reset\" name=\"reset\" value=\"Reset\" />\n";
	print "</form>";
	print "<a href=\"logout_mamullin.pl\">Log Out</a>\n";
	&finish();
# authentication has failed
} else {
	# goto the login page
	print header(-location=>"login_mamullin.pl");
}

# subroutine gets the full catagory word from database
# which only stores three letters
sub getCat() {
	my $rc;
	my $cat = shift @_;
	if( $cat eq "OTH" ) {
		$rc = "OTHER";
	}elsif ( $cat eq "Shp" ) {
		$rc = "Space Ship";
	}elsif ( $cat eq "prb" ) {
		$rc = "Probe";
	}elsif ( $cat eq "bwp" ) {
		$rc = "Balistic Weapon";
	}elsif ( $cat eq "lwp" ) {
		$rc = "Laser/Phaser Weapon";
	}elsif ( $cat eq "cht" ) {
		$rc = "Star Chart";
	}elsif ( $cat eq "stn" ) {
		$rc = "Space Station";
	}elsif ( $cat eq "sat" ) {
		$rc = "Satelite";
	}else { $rc = "UNKNOWN"; }

	return $rc;
}